### PR TITLE
feat: samples for upcoming Airflow tutorial

### PR DESCRIPTION
**Pull request by TheRealSpaceShip**
These are Airflow tutorial-related files.

The tutorial will describe the process of deploying the Airflow platform on GKE Autopilot with next features:
 - Loading DAG files from the GCS bucket by using the GCSFuse CSI driver and mounting volume to Airflow pods. 
 - Configured Ingress with SSL Google-managed certificate. 
 - SSO authentication with Google via OpenID Connect.
 - Monitoring of Airflow with GMP (Google-managed Prometheus)

Also, PR contains a simple Python script to generate secrets for Airflow (Web Server Secret Key and Fernet key) and a simple example of a DAG file.

### CODE REVIEW

**Comment by bourgeoisor** [comment]
@TheRealSpaceShip Is there an ETA for this tutorial?

PR looks good at first glance!

**Comment by TheRealSpaceShip** [comment]
@shannonxtreme Could you please provide ETA for this tutorial?

**Comment by shannonxtreme** [comment]
Im not sure @TheRealSpaceShip it's not on my backlog

File: batch/airflow/airflow-sa.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: airflow
+  namespace: airflow
+  annotations:
+    iam.gke.io/gcp-service-account: airflow-dags-reader@lustrous-baton-363720.iam.gserviceaccount.com
```
Comment (bourgeoisor) [inline batch/airflow/airflow-sa.yaml:21]: Is this hardcoded to somewhere? (`lustrous-baton-363720`)

File: batch/airflow/values.yaml [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+      ephemeral-storage: 1Gi
+    requests:
+      cpu: 250m
+      memory: 512Mi
+      ephemeral-storage: 1Gi
+
+postgresql:
+  enabled: false
+
+dags:
```
Comment (bourgeoisor) [inline batch/airflow/values.yaml:272]: Should these be commented out?

File: batch/airflow/example_dag.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
+#
```
Comment (bourgeoisor) [inline batch/airflow/example_dag.py:1]: Is this python script running inside of a container? Do we need a Dockerfile checked in?

File: batch/airflow/secrets/requirements.txt [hunk 4]
```diff
@@ -0,0 +1 @@
+cryptography
```
Comment (bourgeoisor) [inline batch/airflow/secrets/requirements.txt:1]: Would you mind pinning this to a version?

File: batch/airflow/airflow-sa.yaml (hunk 1 above)
Comment (TheRealSpaceShip) [reply batch/airflow/airflow-sa.yaml:21]: It's leftover from testing. I'll fix it.

File: batch/airflow/values.yaml (hunk 2 above)
Comment (TheRealSpaceShip) [reply batch/airflow/values.yaml:272]: No, postgresql is enable by default but in tutorial I'm using CloudSQL instance istead. And dags is to configure persistence volume that will be mounted to Airflow pods.

File: batch/airflow/example_dag.py (hunk 3 above)
Comment (TheRealSpaceShip) [reply batch/airflow/example_dag.py:1]: This is [DAG](https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/dags.html). Airflow runs it inside of special pod (mount volume with this script which should be placed in GCS bucket). So, no need of Dockerfile.

**Comment by melissakendall** [comment]
Please make sure to add Region tags so we can embed these in the docs (we have been inconsistent with doing this, but it is technically a requirement)

See comment here: https://github.com/GoogleCloudPlatform/kubernetes-engine-samples/pull/804#pullrequestreview-1624728019

**Review by bourgeoisor — APPROVED** [review]
LGTM.

Let me know when I can merge.

Please also create a follow-up PR once the tutorial is published to add the URL of the tutorial in the README.

**Comment by minherz** [comment]
@TheRealSpaceShip please rebase and merge

**Comment by TheRealSpaceShip** [comment]
@minherz I have removed `example_dag.py` file because I took this example from the official Airflow repository and it has a different license header that fails `header-check`. I propose to mention in a tutorial that the user needs to download this file with a command like `curl -L https://raw.githubusercontent.com/apache/airflow/main/airflow/example_dags/example_bash_operator.py -o example_dag.py`

**Comment by TheRealSpaceShip** [comment]
@bourgeoisor this PR is done and ready to be merged.

**Pull request merged**